src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/EventMessageReceiverService.java

```diff
@@ -47,7 +47,13 @@ public EventMessageReceiverService(ObjectMapper objectMapper,
 
     public CaseEventMessage handleDlqMessage(String messageId, String message) {
         log.info("Received Case Event Dead Letter Queue message with id '{}'", messageId);
-        return handleMessage(messageId, message, true);
+        if (featureFlagProvider.getBooleanValue(DLQ_DB_INSERT, getUserId(message))) {
+            return handleMessage(messageId, message, true);
+        } else {
+            log.info("Feature flag '{}' evaluated to false. Message not inserted into database",
+                    DLQ_DB_INSERT.getKey());
+        }
+        return null;
```

Comment:
Just did a quick search, but question: Do we use this  CaseEventMessage or null that we return?

Reply:
It is used by the endpoint in CaseEventHandlerController.java used to emulate receiving a message from the consumer

public CaseEventMessage caseEventHandler(@Valid @RequestBody String message,
                                             @PathVariable("message_id") final String messageId,
                                             @RequestParam(value = "from_dlq",
                                                 required = false) final Boolean fromDlq) 
                                                 
It is used for the assertions in CaseEventHandlerControllerTest

Reply:
Thanks :)